cli/report/report.py

```diff
@@ -269,18 +270,110 @@ def file_jira_issues(self) -> list[str]:
             file_attachments = self.get_file_attachments(
                 step_name=pair["failure"]["step"],
             )
-
-            jira_issue = self.jira.create_issue(
+            labels = [
+                self.job_name,
+                pair["failure"]["step"],
+                pair["rule"]["classification"],
+                pair["failure"]["failure_type"],
+            ]
+
+            # Find duplicate bugs
+            duplicate_bugs = self.find_duplicate_bugs(
                 project=project,
-                summary=summary,
-                description=description,
-                issue_type=type,
-                file_attachments=file_attachments,
+                job_name=self.job_name,
+                failed_step=pair["failure"]["step"],
+                failure_type=pair["failure"]["failure_type"],
             )
-            bugs_filed.append(jira_issue.key)
+
+            # If duplicates are found, comment
+            if len(duplicate_bugs) > 0:
+                for bug in duplicate_bugs:
+                    self.add_duplicate_comment(
+                        bug_key=bug,
+                        failed_step=pair["failure"]["step"],
+                        classification=pair["rule"]["classification"],
+                    )
+
+            # If duplicates are not found, file a bug
+            else:
+                jira_issue = self.jira.create_issue(
+                    project=project,
+                    summary=summary,
+                    description=description,
+                    issue_type=type,
+                    file_attachments=file_attachments,
+                    labels=labels,
+                )
+                bugs_filed.append(jira_issue.key)
 
         return bugs_filed
 
+    def find_duplicate_bugs(
+        self,
+        project: str,
+        job_name: Optional[str],
+        failed_step: str,
+        failure_type: str,
+    ) -> list[str]:
+        """
+        Searches Jira for possible duplicate bugs in the given project. If a bug in the same job, with the same failure_type, and the same failed_step, is found a list of duplicate bugs is returned.
```

Comment:
It could be that the 2 failures in same pod could be due to different reasons. But following our conversation about limiting the abstraction to pod level, its fine by me

Reply:
Yes, that is a known limitation. We are stuck between always filing duplicates or having to sometimes manually file a new bug when this duplicate logic thinks a bug is a duplicate, but it isn't really a duplicate. There isn't much we can do without parsing log messages, and that seems a little over-engineered in my opinion: https://redhat-internal.slack.com/archives/C04PK4QPSR1/p1685021903639669